human_info/src/filter.py

```diff
@@ -2,43 +2,85 @@
 Contains filters used in human pose estimation
 
 """
+import copy
+
+def normsq(x, y):
+    return  (x[0] - y[0])**2 + (x[1] - y[1])**2
+
+class Keypoint_Filter:
+    """
+    Filter apllied to all keypoints in pixel coordinates
+    """
+    def __init__(self, k, num_of_continuity_frames, FEATURES, OUTLIER_THRESHOLD):
+        """
+        @param k: (0, 1) The filter constant. Small value -> Slow dynamics, Large value -> More noise.
+        @param max_dist_pixels: (0, ->) The threshold for deviation from previous estimate for discarding is as an outlier.
+        """
+        self.k = k
+        self.num_of_continuity_frames = num_of_continuity_frames
+        self.FEATURES = FEATURES
+        self.OUTLIER_THRESHOLD = OUTLIER_THRESHOLD
+        self.keypoints = None
+
+    def update(self, keypoints_in):
+        keypoints = copy.deepcopy(keypoints_in)
```

Comment:
Why are you doing a deep copy here?

Reply:
It was to avoid having multiple pointers to the same dictionary. I admit it can be done more efficiently and still safe, but the dict is not so large I think it is an issue.

Reply:
I made a new version without excessive deepcopying.

Reply:
Nice